core-web/libs/sdk/experiments/src/lib/sdk-experiments.ts

```diff
@@ -1,14 +1,141 @@
+     * @return {SdkExperiments} - The instance of the SdkExperiments class.
+     */
+    public static getInstance(config: SdkExperimentConfig): SdkExperiments {
+        if (!SdkExperiments.instance) {
+            SdkExperiments.instance = new SdkExperiments(config);
+
+            this.instance.setExperimentData();
+        }
+
+        return SdkExperiments.instance;
+    }
+
+    /**
+     * Retrieves the current debug status.
+     *
+     * @private
+     * @returns {boolean} - The debug status.
+     */
+    getIsDebugActive(): boolean {
+        return this.config.debug;
+    }
+
+    /**
+     * Fetches experiments from the server.
+     *
+     * @private
+     * @returns {Promise<AssignedExperiments>} - The entity object returned from the server.
+     * @throws {Error} - If an HTTP error occurs or an error occurs during the fetch request.
+     */
+    private async getExperimentsFromServer(): Promise<AssignedExperiments> {
+        try {
+            const response: Response = await fetch(`${this.config.server}${API_EXPERIMENTS_URL}`, {
+                method: 'POST',
+                headers: {
+                    Accept: 'application/json',
+                    'Content-Type': 'application/json'
+                }
+            });
+
+            if (!response.ok) {
+                throw new Error(`HTTP error! status: ${response.status}`);
+            }
+
+            const responseJson = (await response.json()) as IsUserIncludedApiResponse;
+            dotLogger(`Experiment data get successfully `, this.getIsDebugActive());
+
+            return responseJson.entity;
+        } catch (error) {
+            throw new Error(`An error occurred while trying to fetch the experiments: ${error}`);
+        }
+    }
+
+    /**
+     * Retrieves experiment data from the server.
+     *
+     * @private
+     * @async
+     * @throws {Error} If an error occurs while loading the experiments.
+     */
+    private async setExperimentData() {
+        try {
+            const experimentAssignedToUser = await this.getExperimentsFromServer();
+            this.persistExperiments(experimentAssignedToUser);
+        } catch (e) {
+            throw Error(`Error persisting experiments to indexDB, ${e}`);
+        }
+    }
+
     /**
-     * Initializes the method.
+     * Persists experiments to the IndexDB.
      *
-     * @returns {string} - A string indicating that the implementation is pending.
+     * @param {AssignedExperiments} entity - The entity containing experiments to persist.
+     * @private
      */
-    init(): string {
-        return 'Pending Implementation';
+    private persistExperiments(entity: AssignedExperiments) {
+        if (entity.experiments.length > 0) {
+            // TODO: Final parsed data will be stored
+            persistData(entity)
+                .then(() => {
+                    dotLogger(`Experiment data stored successfully `, this.getIsDebugActive());
```

Comment:
` dotLogger('Experiment data stored successfully: ' ` just single quote on this?